src/Microsoft.OpenApi/Services/OpenApiParametersComparer.cs

```diff
@@ -75,6 +75,11 @@ public class OpenApiParametersComparer : OpenApiComparerBase<IList<OpenApiParame
                 var targetParameter = targetParameters
                     .FirstOrDefault(param => param.Name == sourceParameter.Name && param.In == sourceParameter.In);
 
+                if (targetParameter == null)
+                {
+                    continue;
+                }
+
                 WalkAndCompare(
                     comparisonContext,
                     i.ToString(),
```

Comment:
>                   i.ToString(), [](start = 1, length = 32)

This one is kinda tricky but I think it's already alright. It makes sense to use the source index here.

Reply:
Update: With the changes as discussed, we will need to use the target index here.

---
In reply to: [221344148](https://github.com/Microsoft/OpenAPI.NET/pull/316#discussion_r221344148) [](ancestors = 221344148)